Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -578,8 +573,42 @@ private static void ReadNotes(IList<MidiEvent> track, Song song, Song.Instrument
             return GuitarMidiNoteNumberToProcessFnMap;
         }
 
-        static void BuildGuitarMidiNoteNumberToProcessFnDict()
+        static IReadOnlyDictionary<string, ProcessModificationProcessFn> GetTextEventProcessDict(Chart.GameMode gameMode)
+        {
+            switch (gameMode)
+            {
+                case Chart.GameMode.GHLGuitar:
+                    return GhlGuitarTextEventToProcessFnMap;
+                case Chart.GameMode.Drums:
+                    return DrumsTextEventToProcessFnMap;
+
+                default:
+                    return GuitarTextEventToProcessFnMap;
+            }
+        }
+
+        static void SwitchToDrumsVelocityProcessMap(ref EventProcessParams processParams)
         {
+            if (processParams.instrument != Song.Instrument.Drums)
+            {
+                Debug.LogWarning($"Attempted to apply drums velocity process map to non-drums instrument: {processParams.instrument}");
+                return;
+            }
+
+            // Switch process map to drums velocity process map
+            processParams.noteProcessMap = DrumsMidiNoteNumberToProcessFnMap_Velocity;
+        }
+
+        static IReadOnlyDictionary<int, EventProcessFn> BuildGuitarMidiNoteNumberToProcessFnDict()
+        {
+            var processFnDict = new Dictionary<int, EventProcessFn>()
+        {
+            { MidIOHelper.STARPOWER_NOTE, ProcessNoteOnEventAsStarpower },
+            { MidIOHelper.SOLO_NOTE, (in EventProcessParams eventProcessParams) => {
+                ProcessNoteOnEventAsEvent(eventProcessParams, MidIOHelper.SOLO_EVENT_TEXT, MidIOHelper.SOLO_END_EVENT_TEXT);
+            }},
+        };
+
```

Comment:
Would be good to clean up the spacing on 613 -> 620
```suggestion
        static IReadOnlyDictionary<int, EventProcessFn> BuildGuitarMidiNoteNumberToProcessFnDict()
        {
            var processFnDict = new Dictionary<int, EventProcessFn>()
            {
                { MidIOHelper.STARPOWER_NOTE, ProcessNoteOnEventAsStarpower },
                { MidIOHelper.SOLO_NOTE, (in EventProcessParams eventProcessParams) => 
                    {
                        ProcessNoteOnEventAsEvent(eventProcessParams, MidIOHelper.SOLO_EVENT_TEXT, MidIOHelper.SOLO_END_EVENT_TEXT);
                    }
                },
            };

```

Reply:
This doesn't follow how this kind of dictionary entry is formatted in any of the other parsing files. I'd rather not change those as well if it can be helped.

Reply:
Yeah, that's because the dictionary entry format is just plain wrong, Likely an old editor formatting error. No sense in making the problem worse.

Reply:
The current state is concise and it really isn't causing any problems, but if you *really* want me to then I'll do it. It's just tedious having to go and do all of this when it doesn't matter all that much.

To me personally, all this does is add unnecessary lines and padding.

Reply:
Have you seen the "commit suggestion" and "add suggestion to batch" buttons on these github conversations? I'm perfectly fine with you just hitting those buttons and just doing the commit on github since imo that's a quick and easy way to deal with format requests like this. 
Personally I'm hesitant to let more code in that's following the bad patterns set up in the code as it just worsens the problem, which is why I'm being so anal about it.

Reply:
Can't click on those currently, probably because this suggestion is already marked as outdated.
![image](https://user-images.githubusercontent.com/29052821/188261473-148a4eb2-c010-4e81-81a1-b952e40e099b.png)

I'll do it, but no more please. Save the rest (if any) for a new branch.